indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -205,46 +195,63 @@
-     :_index "1_autocomplete"}))
+     :_index "1_autocomplete"
+     :contains-public-collections public-collection?
+     :permitted-group-ids permitted-group-ids}))
 
 (defn- suggestion-doc
   "Creates elasticsearch docs from a given humanized map"
-  [key-name value-map]
+  [permissions key-name value-map]
   (let [values (->> value-map
                     seq
                     (remove #(s/includes? (name (key %)) "-lowercase")))
-        sk-matcher (re-matcher #"science-keywords" key-name)]
+        sk-matcher (re-matcher #"science-keywords" key-name)
+        public-collection? (if (some #(= % "guest") permissions)
+                               true
+                               false)
+        permitted-group-ids (->> permissions
+                                 (remove #(or (= "registered" %) (= "guest" %)))
+                                 (s/join ",")
+                                 not-empty)]
     (if (seq (re-find sk-matcher))
-      (science-keywords->elastic-docs value-map)
-      (map (fn [value]
-             (let [v (val value)
-                   type (camel-snake-kebab/->snake_case_keyword key-name)
-                   id (-> (s/lower-case v)
-                          (str "_" type)
-                          hash)]
-               {:type type
-                :_id id
-                :value v
-                :fields v
-                :_index "1_autocomplete"}))
-           values))))
+     (science-keywords->elastic-docs value-map public-collection? permitted-group-ids)
+     (map (fn [value]
+            (let [v (val value)
+                  type (-> key-name
+                           camel-snake-kebab/->snake_case_keyword
+                           (s/replace #"_humanized|:" ""))
+                  id (-> (s/lower-case v)
+                         (str "_" type)
+                         hash)]
+             {:type type
+              :_id id
+              :value v
+              :fields v
+              :_index "1_autocomplete"
+              :contains-public-collections public-collection?
+              :permitted-group-ids permitted-group-ids}))
+         values))))
 
 (defn- get-suggestion-docs
   "Given the humanized fields from a collection, assemble elastic doc for each
   value available for indexing into elasticsearch"
   [humanized-fields]
-  (for [humanized-field humanized-fields
-        :let [key (key humanized-field)
-              key-name (-> key
-                           name
-                           (s/replace #"([_\.]?humanized(_?2)?|-sn|-id)" ""))
-              value-map (as-> humanized-field h
-                          (val h)
-                          (map util/remove-nil-keys h)
-                          (map #(dissoc % :priority) h))
-              suggestion-docs (map #(suggestion-doc key-name %)
-                                   value-map)]]
-    suggestion-docs))
+  (let [{:keys [permissions]} humanized-fields
+        fields-without-permissions (-> humanized-fields
```

Comment:
change the value to `(dissoc humanized-fields :id :permissions)`

Reply:
Fixed - good catch